gcc4 detection is used only for -fvisibility=hidden, which is ELF-specific

--- origsrc/physfs-2.0.0/CMakeLists.txt	2009-03-23 00:13:28.000000000 -0500
+++ src/physfs-2.0.0/CMakeLists.txt	2010-01-26 03:55:32.226258500 -0600
@@ -52,7 +52,7 @@ IF(CMAKE_COMPILER_IS_GNUCC)
     ENDIF(NOT BEOS)
 
     CHECK_C_SOURCE_COMPILES("
-        #if ((defined(__GNUC__)) && (__GNUC__ >= 4))
+        #if ((defined(__GNUC__)) && (__GNUC__ >= 4)) && !defined(__CYGWIN__)
         int main(int argc, char **argv) { int is_gcc4 = 1; return 0; }
         #else
         #error This is not gcc4.
